bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TableValueImpl.java

```diff
@@ -719,7 +731,7 @@ private class MultiKeyWrapper extends DefaultKeyWrapper {
             public MultiKeyWrapper() {
                 super();
                 List<Type> keyTypes = new ArrayList<>();
-                Type constraintType = type.getConstrainedType();
+                Type constraintType = ((ReferenceType) tableType.getConstrainedType()).getReferredType();
```

Comment:
Shouldn't we use `TypeUtils.getReferredType()` here?

Reply:
It always returns an array type extended from the `ReferenceType` for multiple keys.
So It will provide the referred type as expected.
```
 Type constraintType = TypeUtils.getReferredType(type.getConstrainedType());
```
This should work as well.
As the `TypeUtils` API contains `if` checks, I used the other API for performance.

Reply:
I'll address in https://github.com/ballerina-platform/ballerina-lang/issues/38689